bin/prepare.js

```diff
@@ -21,67 +29,111 @@ by creating separate databases for them
 and running their local bin/prepare.js scripts.
 `)
 
-async function main () {
-    let project, useHttps
+async function prepare () {
     program.parse()
-    const cliOpts = program.opts()
-
-    if (cliOpts.https) {
-        project = undefined
-        useHttps = 'app.localhost'
-    }
-
-    if (!project) project = 'local'
-
-    const secure = Boolean(useHttps)
-    const localhostDomain = (useHttps) ? useHttps : 'localhost'
-    let domains = `127.0.0.1 ${localhostDomain}`
+    const { https, filter } = program.opts()
 
+    // Step 1. Sanity checks
     await checkDockerComposePostgresIsRunning()
-    if (secure) await checkMkCertCommandAndLocalCerts(KEY_FILE, CERT_FILE, useHttps)
-
-    let apps = await getAllActualApps()
-    const appsIndexes = Object.assign({}, ...apps.map((appName, idx) => ({ [appName]: idx + 1 })))
-    if (cliOpts.filter) {
-        apps = apps.filter(app => cliOpts.filter.includes(app))
+    if (https) await checkMkCertCommandAndLocalCerts(KEY_FILE, CERT_FILE, DEFAULT_APP_HTTPS_SUBDOMAIN)
+
+    // Step 2. Get list of available apps and assign ports / indexes / dbNames and so on
+    const allApps = await getAllActualApps()
+    const appsWithData = allApps.map((app, idx) => {
+        const appOrder = idx + 1
+        if (app.type === 'KS') {
```

Comment:
where is the application type defined?

Reply:
Inside of `getAllActualApps` 2 lines above